src/AsyncEventSource.cpp

```diff
@@ -185,17 +188,22 @@ void AsyncEventSourceClient::_queueMessage(AsyncEventSourceMessage *dataMessage)
     delete dataMessage;
     return;
   }
-  if(_messageQueue.length() >= SSE_MAX_QUEUED_MESSAGES){
-      ets_printf("ERROR: Too many messages queued\n");
-      delete dataMessage;
-  } else {
+
+  AsyncWebLockGuard l(_lock);
+  {
```

Comment:
Did you mean to put the lock inside the block?  Otherwise, the block is pointless.

Reply:
Yes. I fixed it.